pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -330,11 +382,18 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
 
         model = self.trainer.lightning_module
 
+        if not self._persistent_metrics:
+            for _, module in model.named_modules():
+                if isinstance(module, Metric):
+                    module.persistent(True)
+            self._persistent_metrics = True
+
         checkpoint = {
             'epoch': current_epoch,
             'global_step': global_step,
             'pytorch-lightning_version': pytorch_lightning.__version__,
             'state_dict': self.trainer.accelerator.lightning_module_state_dict(),
+            'loops_state_dict': self.get_loops_state_dict()
```

Comment:
we also have state dict for callbacks and these have a key "callbacks", not "callbacks_state_dict", so we might want to stick with just "loops" for the key name here.

Reply:
Done !